---
title: "Using triwise to study functional diversity of macrophage progenitors"
author: "Wouter Saelens"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Studying functional diversity of macrophage progenitors}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

In this vignette we illustrate the triwise package using expression data from the three main cell populations contributing to adult tissue resident macrophages in mice (yolk-sac macrophages, fetal liver monocytes and bone marrow monocytes) from this [study](http://www.ncbi.nlm.nih.gov/pubmed/26992565).

```{r, message=FALSE}
library(triwise)
library(limma)
data(vandelaar)
```

This dataset contains log2 gene expression values from, among others, the three progenitors (`YS_MF`, `FL_mono` and `BM_mono`) as an `ExpressionSet` object, each with four biological replicates (samples). We first filter this dataset on only those samples in which we are interested (`Eoi_replicates`) and then calculate the average expression in the three biological conditions (`Eoi`).

```{r, message=FALSE}
Eoi_replicates <- vandelaar[, phenoData(vandelaar)$celltype %in% c("YS_MF", "FL_mono", "BM_mono")]
dim(Eoi_replicates)

Eoi <- avearrays(Eoi_replicates, phenoData(Eoi_replicates)$celltype)
dim(Eoi)
colnames(Eoi)
```

# Plotting gene expression of three conditions

We now transform this gene expression matrix to barycentric coordinates. This transformation will in essence reduce the dimensionality of the gene expression data by one, by only retaining the information of differential expression.
```{r}
barycoords = transformBarycentric(Eoi)
str(barycoords)
```

These barycentric coordinates can then be plotted in a 2D dotplot. In this plot every gene is represented by a single dot. The direction of a dot indicates in which condition(s) the gene is upregulated anThe differentially expressed genes, while the distance from the origin represents the strength of upregulation (in the same order of magnitude as a log fold-change). The differentially expressed genes are shown in black.
```{r, fig.width=8, fig.height=5}
plotDotplot(barycoords)
```
Overall, the main differences in expression seem to be between the two monocytes (`FL_mono` and `BM_mono`) and the yolk sac macrophage (`YS_MF`). However, one can also appreciate the number of genes which are specifically upregulated in one of the two monocytes, or are shared between one monocyte and the yolk sac macrophage.

We can use limma to determine which genes are differentially expressed in any of the three conditions and visualize these genes on the dotplot.
```{r}
design = as(phenoData(Eoi_replicates), "data.frame")
design$celltype = factor(as.character(design$celltype))
design <- model.matrix(~0+celltype, design)
fit <- lmFit(Eoi_replicates, design)
fit = contrasts.fit(fit, matrix(c(1, -1, 0, 0, 1, -1, 1, 0, -1), ncol=3))
fit = eBayes(fit)

top = topTable(fit, p.value=0.05, lfc=1, number=Inf, sort.by = "none")
Gdiffexp = rownames(top)
```

```{r, fig.width=8, fig.height=5}
plotDotplot(barycoords, Gdiffexp)
```

We can also plot this as a rose plot, a polar histogram.
```{r}
plotRoseplot(barycoords, Gdiffexp)
```

The dotplots can also be explored interactively using an htmlwidget. These can be saved as standalone html pages using `saveWidget(dotplot, file="dotplot.html")`.
```{r, fig.width=8, fig.height=5}
dotplot = interactiveDotplot(Eoi, Gdiffexp, Glabels = featureData(vandelaar)$symbol)
dotplot
```


# Analyzing functional diversity of three biological conditions

We will now test whether genes annotated with a particular function through GO are significantly more upregulated in a particular direction than can be expected by chance.
```{r}
library(org.Mm.eg.db)
library(GO.db)
library(dplyr)
gsets = as.list(org.Mm.egGO2ALLEGS)
gsets = sapply(gsets, function(gset) unique(as.character(gset)))
gsetindex = dplyr::bind_rows(lapply(as.list(GOTERM[names(gsets)]), function(goinfo) {
  list(name=Term(goinfo), definition=Definition(goinfo), ontology=Ontology(goinfo), gsetid = GOID(goinfo))
}))
```

```{r}
scores = testUnidirectionality(barycoords, gsets, Gdiffexp, statistic = "diffexp", mc.cores = 8)
scores = left_join(scores, gsetindex, by="gsetid")

redundancy = estimateRedundancy(scores, gsets, Gdiffexp)
scores$redundancy = redundancy[scores$gsetid]
scores = scores[scores$redundancy < 0.2, ]
```

```{r}
plotPvalplot(scores)
interactivePvalplot(scores, gsetindex$name, colnames(Eoi))

plotDotplot(barycoords, Gdiffexp, gsets[["GO:0046697"]])
```

# Visualizing changes in differential expression between pairs of biological conditions

```{r}
diffexp <- function(E_replicates, c1, c2) {
  design = as(phenoData(E_replicates)[phenoData(E_replicates)$celltype %in% c(c1,c2),], "data.frame")
  design$celltype = factor(as.character(design$celltype), levels=c(c2, c1)) # relevel so that contrast is in the right direction
  designmat = model.matrix(~0+celltype, design)
  colnames(designmat) <- c("A","B")
  fit <- lmFit(E_replicates[,rownames(design)[design$celltype %in% c(c1,c2)]],designmat)
  contrastmat <- makeContrasts("B-A", levels=designmat)
  fit = contrasts.fit(fit, contrastmat)
  fit = eBayes(fit)
  top = topTable(fit, p.value=0.1, lfc=1, number=Inf, sort.by = "none")
  
  list(up = rownames(top)[top$logFC > 1], down=rownames(top)[top$logFC < -1])
}

Goi = list(
  YSup = intersect(diffexp(vandelaar, "YS_MF_AMF", "FL_mono_AMF")$up, diffexp(vandelaar, "YS_MF_AMF", "BM_mono_AMF")$up),
  YSdown = intersect(diffexp(vandelaar, "YS_MF_AMF", "FL_mono_AMF")$down, diffexp(vandelaar, "YS_MF_AMF", "BM_mono_AMF")$down),
  FLup = intersect(diffexp(vandelaar, "FL_mono_AMF", "BM_mono_AMF")$up, diffexp(vandelaar, "FL_mono_AMF", "YS_MF_AMF")$up),
  FLdown = intersect(diffexp(vandelaar, "FL_mono_AMF", "BM_mono_AMF")$down, diffexp(vandelaar, "FL_mono_AMF", "YS_MF_AMF")$down),
  BMup = intersect(diffexp(vandelaar, "BM_mono_AMF", "FL_mono_AMF")$up, diffexp(vandelaar, "BM_mono_AMF", "YS_MF_AMF")$up),
  BMdown = intersect(diffexp(vandelaar, "BM_mono_AMF", "FL_mono_AMF")$down, diffexp(vandelaar, "BM_mono_AMF", "YS_MF_AMF")$down)
)
Goi = Filter(function(x) length(x>0), Goi)
```

```{r, fig.width=8, fig.height=5}
Eoi2_replicates <- vandelaar[, phenoData(vandelaar)$celltype %in% c("YS_MF_AMF", "FL_mono_AMF", "BM_mono_AMF")]
Eoi2 <- avearrays(Eoi2_replicates, phenoData(Eoi2_replicates)$celltype)
barycoords2 <- transformBarycentric(Eoi2)

plotDotplot(barycoords2, unlist(Goi), Goi=Goi, barycoords2 = barycoords)
```


